other/upgrade_2-1_mysql.sql

```diff
@@ -2347,8 +2347,12 @@ if ($doChange)
 ALTER TABLE {$db_prefix}log_floodcontrol ADD COLUMN ip VARBINARY(16);
 ---#
 
+---# Modify log_type size
+ALTER TABLE {$db_prefix}log_floodcontrol MODIFY log_type VARCHAR(30) NOT NULL DEFAULT 'post';
+---#
+
 ---# Create primary key for floodcontrol
-ALTER TABLE {$db_prefix}log_floodcontrol ADD PRIMARY KEY (ip,log_type);
+ALTER TABLE {$db_prefix}log_floodcontrol ADD PRIMARY KEY (ip(16),log_type(30));
```

Comment:
nooooooo,
first ip is no 16 long
secondly remove the (30) from the pk definition and do the same on install part.